library/std/src/thread/scoped.rs

```diff
@@ -35,28 +39,133 @@ pub struct Scope<'scope, 'env: 'scope> {
 #[stable(feature = "scoped_threads", since = "1.63.0")]
 pub struct ScopedJoinHandle<'scope, T>(JoinInner<'scope, T>);
 
+const WAITING_BIT: usize = 1;
+const ONE_RUNNING: usize = 2;
+
+/// Artificial limit on the maximum number of concurrently running threads in scope.
+/// This is used to preemptively avoid hitting an overflow condition in the running thread count.
+const MAX_RUNNING: usize = usize::MAX / 2;
+
+#[derive(Default)]
 pub(super) struct ScopeData {
-    num_running_threads: AtomicUsize,
-    a_thread_panicked: AtomicBool,
-    main_thread: Thread,
+    sync_state: AtomicUsize,
+    thread_panicked: AtomicBool,
```

Comment:
I think the previous name makes more sense. This is true when any thread panicked. `thread_panicked` to me suggests that there is only a single thread.